src/cbmc_starter_kit/setup_ci.py

```diff
@@ -42,6 +48,14 @@ def parse_arguments():
                 run CBMC proofs in GitHub Actions. If your repo has a large
                 runner available, you can specify it here. default: %(default)s"""
         }, {
+        'flag': '--aws-account-id',
+        'type': lambda x: x if x.isdigit() and len(x) == 12 else False,
```

Comment:
> it's basically going to ignore the user's input if it isn't a number (because you check if args.aws_account_id below). Maybe I don't understand but I think this is a bit confusing? If the user gives an invalid command line arg the usual behavior is to exit?

If the user doesn't provide a value that is a 12-digit string, then the command will exit.

> Why not write a proper function that raises ValueError so that the argparser can handle it and print an error message?

You don't have to. The use of an anonymous function here is actually the most concise option in terms of handing invalid user input for this argument.

Reply:
I was wrong. I have updated the `type` to be a function that will raise ArgumentTypeError if the provided input is not a 12-character string consisting only of digits.